Library/Homebrew/test/utils/git_spec.rb

```diff
@@ -0,0 +1,209 @@
+require "utils/git"
+
+describe Git do
+  before(:all) do
+    git = HOMEBREW_SHIMS_PATH/"scm/git"
+    file = "lib/blah.rb"
+    repo = Pathname.new("repo")
+
+    (repo/"lib").mkpath
+    system "#{git} init"
+    FileUtils.touch("repo/#{file}")
+
+    File.open(repo/file, "w") { |f| f.write("blah") }
+    system git, "add", repo/file
+    system git, "commit", "-m", "'File added'"
+    @h1 = `git rev-parse HEAD`
+
+    File.open(repo/file, "w") { |f| f.write("brew") }
+    system git, "add", repo/file
+    system git, "commit", "-m", "'written to File'"
+    @h2 = `git rev-parse HEAD`
+  end
+
+  let(:file) { "lib/blah.rb" }
+  let(:repo) { Pathname.new("repo") }
+  let(:hash1) { @h1[0..6] }
+  let(:hash2) { @h2[0..6] }
+
+  after(:all) do
+    FileUtils.rm_rf("repo")
+  end
+
+  describe "#last_revision_commit_of_file" do
+    it "sets args as --skip=1 when before_commit is nil" do
+      expect(
+        described_class.last_revision_commit_of_file(repo, file),
+      ).to eq(hash1)
+    end
+
+    it "sets args as --skip=1 when before_commit is nil" do
+      expect(
+        described_class.last_revision_commit_of_file(repo,
+                                                    file,
+                                                    before_commit: "0..3"),
+      ).to eq(hash2)
+    end
+  end
+
+  describe "#last_revision_of_file" do
+    it "returns last revision of file" do
+      expect(
+        described_class.last_revision_of_file(repo,
+                                              repo/file),
+      ).to eq("blah")
+    end
+
+    it "returns last revision of file based on before_commit" do
+      expect(
+        described_class.last_revision_of_file(repo, repo/file,
+                                              before_commit: "0..3"),
+      ).to eq("brew")
+    end
+  end
+end
+
+describe Utils do
+  describe "::git_available?" do
+    it "returns true if git --version command succeeds" do
```

Comment:
This test is behaving weird on my system. It passes sometimes and fails at other times.
Basically, `git_available?` is returning `nil` and `false` too at some times.